Feat/lint using super json

Jakub-Vacek opened the pull request:
<!--- Tip: You don't have to remove these comments -->

## Description
<!--- Describe your changes in detail -->
This PR improves DX of Lint command 

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!--- If your changes are only to the internals then make sure any function has its attached documentation updated or added (e.g. descriptive name, doc comment, etc.). If your changes are to user-facing APIs or concepts then make sure README.md is updated accordingly (e.g. command help output). -->
- [x] I have updated the documentation accordingly. For updating Oclif commands documentation use [oclif-dev](https://github.com/oclif/dev-cli#oclif-dev-readme).
- [x] I have read the **CONTRIBUTION_GUIDE** document.
- [x] I haven't repeated the code. (DRY)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

lukas-valenta commented on src/logic/lint.ts:
```diff
@@ -305,6 +154,7 @@ export function formatHuman(
   } else {
     buffer += formatIssues(report.errors);
 
+    console.log('buf', buffer);
```
nope

Jakub-Vacek replied:
//bo 😅

lukas-valenta reviewed (approved):
Looks good

lukas-valenta commented on src/test/utils.ts:
```diff
@@ -323,10 +324,14 @@ export async function execCLI(
 
-    childProcess.on('error', (err: Error) => reject(err));
+    childProcess.on('error', (err: Error) => {
+      if (rejectOnErr) reject(err);
```
except this, this not looks good, i missed that

Jakub-Vacek replied:
Yep, I missed that too. I was playing with errors in child process and forget to discard my changes.

Jakub-Vacek commented:
@lukas-valenta Do you want to give it a quick look? I added lint integration tests where I had to fiddle with stdout mocking and exposing it from child process I also added hidden flag to skip map to profile validation - we use lint in a lot of our tests.

lukas-valenta commented on src/logic/lint.ts:
```diff
@@ -319,20 +320,21 @@ async function prepareLintedMap(
     ast: mapAst,
-    path,
+    path:
+      map.path ||
```
```suggestion
      map.path ??
```
